console/src/view/mod.rs

```diff
@@ -0,0 +1,61 @@
+use crate::input;
+use tui::layout;
+
+mod task;
+
+pub(crate) enum View {
+    /// The table list of all tasks.
+    TasksList,
+    /// Inspecting a single task instance.
+    TaskInstance(self::task::TaskView),
+}
+
+impl View {
+    pub(crate) fn update_input(&mut self, event: input::Event, tasks: &mut crate::tasks::State) {
+        match self {
+            View::TasksList => {
+                // The enter key changes views, so handle here since we can
+                // mutate the currently selected view.
+                match event {
+                    input::Event::Key(input::KeyEvent {
+                        code: input::KeyCode::Enter,
+                        ..
+                    }) => {
+                        if let Some(task) = tasks.selected_task().upgrade() {
+                            *self = View::TaskInstance(self::task::TaskView::new(task));
+                        }
+                    }
+                    _ => {
+                        // otherwise pass on to view
+                        tasks.update_input(event);
+                    }
+                }
+            }
+            View::TaskInstance(view) => view.update_input(event),
+        }
+    }
+
+    pub(crate) fn render<B: tui::backend::Backend>(
+        &mut self,
+        frame: &mut tui::terminal::Frame<B>,
+        area: layout::Rect,
+        tasks: &mut crate::tasks::State,
+    ) {
+        match self {
+            View::TasksList => {
+                tasks.render(frame, area);
+                tasks.retain_active();
+            }
+            View::TaskInstance(view) => {
+                view.render(frame, area);
+                // retain_active()? should that always be done?
```

Comment:
`retain_active` drops data stored for completed tasks after a period of time...we probably want to run that here so that we don't use increasingly large amounts of memory to store data for tasks that have ended?

since the task view clones the `Rc` for the task that's being actively looked at, we won't yank it out from under the user if we continue deleting completed tasks in the background...